cmake/presets/clang.cmake

```diff
@@ -2,7 +2,8 @@
 
 set(CMAKE_CXX_COMPILER "clang++" CACHE STRING "" FORCE)
 set(CMAKE_C_COMPILER "clang" CACHE STRING "" FORCE)
-set(CMAKE_CXX_FLAGS "-Wall -Wextra -g -O2 -DNDEBUG" CACHE STRING "" FORCE)
+set(CMAKE_CXX_FLAGS_DEBUG "-Wall -Wextra -g" CACHE STRING "" FORCE)
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-Wall -Wextra -g -O2 -DNDEBUG" CACHE STRING "" FORCE)
```

Comment:
Is there a reason to overwrite those? Couldn't you just use `set(CMAKE_CXX_FLAGS "-Wall -Wextra")` as `CMAKE_CXX_FLAGS` is used regardless of the value of `CMAKE_BUILD_TYPE`, then it would apply to all build types.

Reply:
yes. they make the most sense with "debug" configurations only and can be distracting otherwise.

Reply:
Ok, good point if we ever fix all the warnings, we can move it back. 

I like `-Wpedantic -Wshadow -Wconversion` as well as they are no included in `-Wextra`.